code/game/objects/items/toys.dm

```diff
@@ -1413,4 +1414,34 @@ obj/item/toy/cards/deck/syndicate/black
 
 
 /obj/item/toy/russian_revolver/proc/spin_cylinder()
-	bullet_position = rand(1,6)
\ No newline at end of file
+	bullet_position = rand(1,6)
+
+/*
+ * Rubber Chainsaw
+ */
+/obj/item/weapon/twohanded/toy/chainsaw
+	name = "Toy Chainsaw"
+	desc = "A toy chainsaw with a rubber edge. Ages 8 and up"
+	icon_state = "chainsaw0"
+	force = 0
+	throwforce = 0
+	throw_speed = 4
+	throw_range = 20
+	wieldsound = 'sound/weapons/chainsawstart.ogg'
+	attack_verb = list("sawed", "cut", "hacked", "carved", "cleaved", "butchered", "felled", "timbered")
+	
+obj/item/weapon/twohanded/toy/chainsaw/update_icon()
+	if(wielded)
+		icon_state = "chainsaw[wielded]"
+	else
+		icon_state = "chainsaw0"
+
+/obj/item/weapon/twohanded/toy/chainsaw/attack(mob/target as mob, mob/living/user as mob)
+	if(wielded)
+		playsound(loc, 'sound/weapons/chainsaw.ogg', 100, 1, -1)
+		..()
+		return
+	else
+		playsound(loc, "swing_hit", 50, 1, -1)
+		..()
+		return

```

Comment:
There's a bit of redundancy here, starting from line 1440, this could just be replaced with the following:
```
if(wielded)
    playsound(loc, 'sound/weapons/chainsaw.ogg', 100, 1, -1)
else
    playsound(loc, "swing_hit", 50, 1, -1)
..()
return
```